src/hooks/useBalance.ts

```diff
@@ -0,0 +1,89 @@
+// Copyright 2021 Parity Technologies (UK) Ltd.
+// This file is part of Parity Bridges UI.
+//
+// Parity Bridges UI is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Parity Bridges UI is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Parity Bridges UI.  If not, see <http://www.gnu.org/licenses/>.
+import { Balance } from '@polkadot/types/interfaces';
+import { formatBalance } from '@polkadot/util';
+import BN from 'bn.js';
+import { useEffect, useState } from 'react';
+
+import { useSourceTarget } from '../contexts/SourceTargetContextProvider';
+
+type State = [string, Balance, boolean, string];
+
+const ZERO = new BN(0);
+
+const useBalance = (address: string, providedSi: boolean = false): State[] => {
+  const [sourceState, setSourceState] = useState<State>(['0', new BN(ZERO) as Balance, true, '-']);
+  const [targetState, setTargetState] = useState<State>(['0', new BN(ZERO) as Balance, true, '-']);
+
+  const {
+    sourceChainDetails: {
+      sourceApiConnection: { api: sourceApi }
+    },
+    targetChainDetails: {
+      targetApiConnection: { api: targetApi }
+    }
+  } = useSourceTarget();
+
+  useEffect((): (() => void) => {
+    let unsubscribeSource: null | (() => void) = null;
+    let unsubscribeTarget: null | (() => void) = null;
+    if (address) {
+      sourceApi.query.system
+        .account(address, ({ data }): void => {
+          setSourceState([
+            formatBalance(data.free, {
+              decimals: sourceApi.registry.chainDecimals[0],
+              forceUnit: '-',
+              withSi: providedSi
```

Comment:
We need proper units and human-readable formatting here.

Reply:
I would like to discuss this a bit more in order to understand

Reply:
So I'm not sure exactly how `formatBalance` works and perhaps we are already doing what  I want to achieve.
I'd like the values to be displayed like that:
```
1 TOK
1 kTOK
1 MTOK
1 mTOK
```
etc. So using SI prefixes and unit symbol - i.e avoiding a situation where full number of decimals is being displayed.
My suggestion would be also to display full balance (no SI; with thousand separator) in a tooltip (`title` attribute is good enough).

Reply:
Discussed this offband and the API is doing exactly what I mean, the unit should be loaded from the chainspec's `properties` section and currently it's missing for Rialto & Millau, hence it displays incorrectly, but we will add it shortly.

Reply:
> display full balance (no SI; with thousand separator) in a tooltip (title attribute is good enough).

I'll make this in different PR though (keep scope of PRs small)